woocommerce/woocommerce
# dev: refactor core profiler loader

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Pure refactor that should not change any user facing details, this PR extracts the commonalities of the Loader component that we will be using throughout onboarding.

Closes https://github.com/woocommerce/team-ghidorah/issues/254.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set up WooCommerce and go through the onboarding profiler
2. Loader screen after plugin selection step as well as the skip guided setup step should work as before
3. Run storybook `pnpm run storybook` from within the `tools/storybook` folder and check out the Onboarding\Example Loader story, should look like:

https://github.com/woocommerce/woocommerce/assets/27843274/12af8577-64d0-4080-b4e5-4a28c5bd1604





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
